qemu/tests/qemu_guest_agent.py

```diff
@@ -294,6 +301,34 @@ def gagent_check_install_uninstall(self, test, params, env):
         session.close()
 
     @error_context.context_aware
+    def gagent_check_stop_start(self, test, params, env):
+        """
+        Repeat stop/restart qemu-ga service in guest.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        repeats = int(params.get("repeat_times", 1))
+        logging.info("Repeat stop/restart qemu-ga service for %s times" % repeats)
+
+        if not self.vm:
+            self.vm = self.env.get_vm(params["main_vm"])
+            self.vm.verify_alive()
+        session = self._get_session(params, self.vm)
+        for i in xrange(repeats):
+            error_context.context("Repeat: %s/%s" % (i + 1, repeats),
+                                  logging.info)
+            self.gagent_stop(session, self.vm)
+            time.sleep(1)
+            self.gagent_start(session, self.vm)
+            time.sleep(1)
+        session.close()
+        error_context.context("Check if the guest agent could still work normally",
+                              logging.info)
+        self.gagent_verify(params, self.vm)
```

Comment:
I think this check should be applied for every iteration.

Reply:
Modified.
Pls have a check.